### ISSUE TITLE
Allow using gradient checkpointing

### ISSUE THREAD

**Issue by tvercaut** [NifTK/NiftyNet#325]
As per discussion with @stian-johnsen, it might be relatively simple to allow the use of gradient checkpointing in NiftyNet.

This could be made into an option in the config file to expose the option to the user.

Relevant link:
- https://github.com/openai/gradient-checkpointing

**Comment by stian-johnsen** [comment]
On a technical level, the integration is indeed quite simple. However, in practice, the module is of very limited use since automatic checkpoint extraction failed for me on all complex networks I tried, and only worked on the module authors' test networks. Manual checkpoint specification would be an option, but seeing how individual Tensorflow operations must be specified, I very much doubt it would see much use, and be worth the added code and resulting maintenance overhead. 
An FAQ entry reiterating these findings has been made.

**Comment by stian-johnsen** [comment]
An FAQ summarising the issues with gradient checkpointing in NiftyNet has been made here: https://github.com/NifTK/NiftyNet/wiki/NiftyNet-FAQ#using-niftynet-with-gradient-checkpointing
